File: binarySearch/binarySearch.js
// Space: O(1)
// Time: O(log(n))
var search = function(nums, target) {
    let s = 0, e = nums.length-1;

    while (s <= e) {
        let m = Math.floor((e - s) / 2) + s

        if (nums[m] === target) {
            return m
        } else if (nums[m] < target) {
            s = m + 1
        } else {
            e = m - 1
        }
    } 

    return -1;
};
console.log(search([-1,0,3,5,9,12], 5)) // = 3

File: spaced_rep_java_oct_2024/binarySearch.java
package spaced_rep_java_oct_2024;

class Solution {
    public int search(int[] nums, int target) {
        int low = 0; int high = nums.length-1;

        while (low <= high) {
            int m = ((int) Math.floor((high+low) / 2) );
      

            if (nums[m] == target)             {
                return m;
            } else if (target < nums[m]) {
                high = m -1;
            } else {
                low = m+1;
            }
        }

        return -1;
    }
    public static void main(String[] args) throws Exception {
        containsDuplicate sol = new containsDuplicate();

        System.out.println("value: " + sol.search(new int[]{-2,0,3,5,9,12}, 12));
    }
}